core/rootobject/collection.go

```diff
@@ -0,0 +1,294 @@
+			return mro.TheirRootObj, &merge.MergeStats{
+				Operation:            merge.TableAdded,
+				Adds:                 0,
+				Deletes:              0,
+				Modifications:        0,
+				DataConflicts:        0,
+				SchemaConflicts:      0,
+				ConstraintViolations: 0,
+			}, nil
+		case mro.TheirRootObj == nil && mro.AncestorRootObj != nil:
+			return nil, &merge.MergeStats{
+				Operation:            merge.TableRemoved,
+				Adds:                 0,
+				Deletes:              0,
+				Modifications:        0,
+				DataConflicts:        0,
+				SchemaConflicts:      0,
+				ConstraintViolations: 0,
+			}, nil
+		case mro.TheirRootObj == nil && mro.AncestorRootObj == nil:
+			return nil, &merge.MergeStats{
+				Operation:            merge.TableUnmodified,
+				Adds:                 0,
+				Deletes:              0,
+				Modifications:        0,
+				DataConflicts:        0,
+				SchemaConflicts:      0,
+				ConstraintViolations: 0,
+			}, nil
+		default:
+			return nil, nil, errors.New("HandleMerge has somehow reached a default case")
+		}
+	} else if mro.TheirRootObj == nil {
+		switch {
+		case mro.AncestorRootObj != nil:
+			return nil, &merge.MergeStats{
+				Operation:            merge.TableModified,
+				Adds:                 0,
+				Deletes:              0,
+				Modifications:        0,
+				DataConflicts:        1,
+				SchemaConflicts:      0,
+				ConstraintViolations: 0,
+			}, nil
+		case mro.AncestorRootObj == nil:
+			return mro.OurRootObj, &merge.MergeStats{
+				Operation:            merge.TableAdded,
+				Adds:                 0,
+				Deletes:              0,
+				Modifications:        0,
+				DataConflicts:        0,
+				SchemaConflicts:      0,
+				ConstraintViolations: 0,
+			}, nil
+		default:
+			return nil, nil, errors.New("MergeRootObjects has somehow reached a default case")
+		}
+	}
+	identifier := mro.OurRootObj.(RootObject).GetID()
+	if int64(identifier) >= int64(len(globalCollectionFunctions)) {
+		return nil, nil, errors.New("unsupported root object found, please upgrade Doltgres to the latest version")
+	}
+	collFunc := globalCollectionFunctions[identifier]
+	if collFunc == nil {
+		return nil, nil, errors.Newf("invalid root object found, ID: %d", int64(identifier))
+	}
+	return collFunc.HandleMerge(ctx, mro)
+}
+
+// LoadAllCollections loads and returns all collections from the root.
+func LoadAllCollections(ctx context.Context, root RootValue) ([]Collection, error) {
+	colls := make([]Collection, 0, len(globalCollectionFunctions))
+	for _, collFuncs := range globalCollectionFunctions {
+		if collFuncs == nil {
+			continue
+		}
+		coll, err := collFuncs.LoadCollection(ctx, root)
+		if err != nil {
+			return nil, err
+		}
+		colls = append(colls, coll)
+	}
+	return colls, nil
+}
+
+// LoadCollection loads the collection matching the given ID from the root.
+func LoadCollection(ctx context.Context, root RootValue, collectionID RootObjectID) (Collection, error) {
```

Comment:
Ideally this is replaced by 3 methods in the new package, i.e. `GetSequences`, `GetUserTypes`, `GetUserFunctions`.

It's also possible it's not necessary, and all code can refer to the particular collection package to load from a RootValue